# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../hotplug++/hotplug++-linux-2.6.18-no-isapnp-header.patch
# Copyright (C) 2006 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

--- ./bin/isapnp-match.hh.orig	2006-09-29 14:36:49.000000000 +0200
+++ ./bin/isapnp-match.hh	2006-09-29 14:45:10.000000000 +0200
@@ -2,8 +2,19 @@
 #include <vector>
 #include <match.hh>
 
+/*
+ if we use the sanitized linux headers >= 2.6.18 we will not have isapnp.h
+ so we have to define ISAPNP_VENDOR and ISAPNP_DEVICE by ourself.
+ */
+#define ISAPNP_VENDOR(a,b,c)    (((((a)-'A'+1)&0x3f)<<2)|\
+		((((b)-'A'+1)&0x18)>>3)|((((b)-'A'+1)&7)<<13)|\
+		((((c)-'A'+1)&0x1f)<<8))
+#define ISAPNP_DEVICE(x)    ((((x)&0xf000)>>8)|\
+		(((x)&0x0f00)>>8)|\
+		(((x)&0x00f0)<<8)|\
+		(((x)&0x000f)<<8))
+
 #define ISAPNP_CARD_DEVS	8
-#include <linux/isapnp.h>
 
 class isapnp_module {
 public:
